src/nock/recorder.ts

```diff
@@ -282,10 +314,125 @@ export async function canUpdateTraffic(
 
   if (
     updateTraffic &&
-    (await exists(composeRecordingPath(recordingPath, 'new')))
+    (await exists(composeRecordingPath(recordingPath, { version: 'new' })))
   ) {
     return true;
   }
 
   return false;
 }
+
+export async function decodeRecordings(
+  recordings: RecordingDefinition[]
+): Promise<RecordingDefinition[]> {
+  return Promise.all(recordings.map(decodeRecordingResponse));
```

Comment:
Promise.all have problems when number of promises is higher then 10. This should be resolved in TS 4.5.0 but I didn't test it. I just found the limitations when working on DX scanner. 

https://stackoverflow.com/questions/70062994/why-an-array-of-more-than-10-elements-in-promise-all-generates-the-problem-no-o